awx/main/isolated/manager.py

```diff
@@ -191,6 +192,14 @@ def check(self, interval=None):
 
             last_check = time.time()
 
+        if self.check_callback is not None:
+            config_path = self.path_to('artifacts', self.ident, 'config')
+            # If the configuration artifact has been synced back, update the model
+            if os.path.exists(config_path):
+                with open(config_path, 'r') as f:
+                    data = json.load(f)
+                self.check_callback(data)
```

Comment:
This still may be called multiple times, causing a database update each time it's called. We only need to update the database record once. Some sort of `self.called_check_callback_flag` here or in the callback registered outside would ensure that only 1 database update is called.

Reply:
This looks good.

Reply:
We want to save this invocation information as early as possible. Waiting for the job to finish is late. The information that we are saving is vital to debugging/recreating errors. Maybe check for the artifact on each loop and set a flag when you have processed that file and called the check_callback that was passed in?

Reply:
Yeah, fair enough.  I'll come up with something.

Reply:
Got it, I think.